### ISSUE TITLE
Redundant null check causes nullability warning

### ISSUE THREAD

**Issue by krwq** [dotnet/roslyn#44834]
Redundant check is causing unnecessary warnings by the compiler

**Version Used**: 
master from June 1

**Steps to Reproduce**:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8DEA7AVwwwmAzgAI49TyBYAKAAEAmARkaYGYLWKBhCgG9GFMbx5MALBQAqcWAAombAAwB+CgDMA9joCUo8SIbizFNFoqLdOigF57FQsUOnzYkx4+2HvNQB0AKJgAA4wAJ4A3EYeAL6x5olmlta+AIROLhj6FAD0ec52AO4QCHhoeADmFMUAFtQUMHVoUBZtCHBgOgBucAAmycZDYgnuHiMWVjZ6AQAy1FXNDk6qbt7Ck2NmY3FAA===

```csharp
    public void Test(string? foo)
    {
        if (foo == null)
        {
            foo = string.Empty;
        }
        
        // redundant check causes warning
        if (foo != null) // no warning when this is removed
        {
        }
        
        if (foo.Length == 0)
        {
        }
    }
```

**Expected Behavior**:

No warning

**Actual Behavior**:

`warning CS8602: Dereference of a possibly null reference.`

**Comment by gafter** [comment]
This is by design. We take an explicit null check in the code to be a sign that the user expects the value might haven been null.

**Comment by krwq** [comment]
@gafter we've just chatted about this and I don't seem the only one confused by this - would it make sense to warn that check is redundant instead? (at least for the obvious cases)
